app/controllers/items_controller.rb

```diff
@@ -1,42 +1,45 @@
 class ItemsController < ApplicationController
   #This controller's function is for the items that are stored in the library database. It has operations that create, delete,
   #and edit the items
-  
+
   #before_action :current_user.admin?         #WIP must verify that the user is an admin in order for to run any procudure in this controller. An admin
-                                              #should be the only user that is able to add/edit/delete items in the database
-  
+  #should be the only user that is able to add/edit/delete items in the database
+
+
   def index                                   #items in library
     @items = search
-    @page_title = "All library resources"
-    
-    if @items.nil?
-      # case when we don't use search, just list all
-      @items = Item.all
-      
+
+    if params[:search].blank?
+      @page_title = "All library resources"
     elsif @items.empty?
       # case when there is no result
-      @page_title = "Must type exact title \"#{params[:search]}\":"
-      
+      @page_title = "Sorry, we cannot find any results for \"#{params[:search]}\":"
     else
       # case when we found the result
       @page_title = "Results for \"#{params[:search]}\":"
     end
-  
+
+    @pendings, @approveds = [], []
+
+    @items.each do |item|
+      @pendings << item if item.status == Item::PENDING
```

Comment:
Since `@items` is an activerecord relation, we can get the list of pending and approved items by performing an additional query on it instead of sorting it manually like this, i.e. `@pendings = @items.where(status: Item::PENDING)`

Reply:
Yes, I did try this initially but it seems like Ruby converts `@items` to normal array time. I got an error message saying that the Array does not have `where` function.